unstoppable-ios-app/domains-manager-ios/Services/WalletConnectService_V2/WalletConnectServiceV2.swift

```diff
@@ -75,14 +78,21 @@ class WalletConnectServiceV2: WalletConnectServiceV2Protocol {
     }
     
     // returns both V1 and V2 apps
-    func getConnectedApps() -> [UnifiedConnectAppInfo] {
-        appsStorage.retrieveApps().map{ UnifiedConnectAppInfo(from: $0)}
-            + appContext.walletConnectService.getConnectedAppsV1().map{ UnifiedConnectAppInfo(from: $0)}
+    func getConnectedApps() async -> [UnifiedConnectAppInfo] {
+        let unifiedApps = getAllUnifiedAppsFromCache()
+        
+        // trim the list of connected dApps
+        let validDomains = await appContext.dataAggregatorService.getDomains()
+        let validConnectedApps = unifiedApps.trimmed(to: validDomains)
+        
+        // disconnect those connected to gone domains
+        disconnectApps(from: unifiedApps, notIncluding: validConnectedApps)
+        return validConnectedApps
     }
-    
+        
     func disconnectAppsForAbsentDomains(from domains: [DomainItem]) {
         Task {
-            let connectedApps = getConnectedApps()
+            let connectedApps = await getConnectedApps()
             for app in connectedApps {
                 if domains.first(where: { $0.name == app.domain.name }) == nil {
                     try? await disconnect(app: app)
```

Comment:
Looks like a duplicate of function above now

Reply:
True 👍re-used